### PR TITLE
Override hashCode to avoid reading data

### PR DESCRIPTION
**Pull request by alamar**
fixes ChronicleEnterprise/Chronicle-FIX#838

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/bytes/internal/CommonMappedBytes.java [hunk 1]
```diff
@@ -591,6 +591,12 @@ public String toString() {
                 '}';
     }
 
+    @Override
+    // Avoid reading contents as it may be empty or point to unmapped region.
+    public int hashCode() {
+        return mappedFile.hashCode();
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/bytes/internal/CommonMappedBytes.java:597]: I had not realised that the reference accounting called hashCode(). Why was it crashing before? Were we not pointing to valid memory?

**Review by JerryShea — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/bytes/internal/CommonMappedBytes.java (hunk 1 above)
Comment (minborg) [reply src/main/java/net/openhft/chronicle/bytes/internal/CommonMappedBytes.java:597]: This is the same problem as for ByteBuffer. It is questionable to have hashCode() and equals() implementations like these for memory regions. Make sure equals() is overridden in the same way as hashCode() !

Comment (alamar) [reply src/main/java/net/openhft/chronicle/bytes/internal/CommonMappedBytes.java:597]: I have added equals() @minborg

**Comment by minborg** [comment]
This is a behavioral change. What is the general principle when it comes to equal(), hashcode() for these memory segments? We should apply the same principle for all memory segment types. Perhaps we should have a short meeting about this. 

The current solution is fundamentally flawed and buggy. For example, even though we can use a `long` for addressing, hashing is limited to 2^31 bytes. It is likely common practice to use Bytes as key in a Map<CharSequence, ?>. Calling hashCode on large memory segments is very slow and would have impacts on caching etc. So, many conflicting requirements here.

**Comment by alamar** [comment]
It would indeed be nice to have a short meeting.

**Pull request closed without merging**